components/partials/spot/trading/trade.vue

```diff
@@ -1327,20 +1714,138 @@ export default Vue.extend({
         return
       }
 
-      if (form.amount.trim() !== '') {
+      if (form.amount.trim() !== '' && !form.amount.includes('.')) {
```

Comment:
the logic surrounding the amount input fields is very interesting.  the input fields are of type `number`, but we emit the values as strings to updating `form.amount`, as an example, which is a string. then, if we want to do operations like greater than/less than on `form.amount`, we convert to big number to use its `.gt` or `isLessThan` methods. 

the reason for this check of `form.amount.includes('.')` is that an input field of type number will not emit a `target.value` of (42.), it will only include the decimal place if there are values past the decimal (i.e. 42.2).

so for an input field displaying `42.`, the value emitted to update form.amount would still be `42`, which is the same as `42`, and therefor Vue is not rerendering the component to remove the trailing `.` in `42.`, which is less than ideal from a UX perspective.  so I am adding in the key to refresh the input field, and then set its value to `42`, which destroys the trailing `.`, whoo! was a journey to get there tbh. could be a good thing for me to demo at our UI meeting so we can use the strategy elsewhere in the app if we want

Reply:
Ah I see why you're doing this now, but maybe we should fix this at a deeper level wouldn't you agree? In the Input component itself ideally? I feel like its currently all a bit laborious to get stuff like this working. Maybe if implementing this in the Input component yields trouble we can pass a flag that tells it to do this and default the flag to false?